src/NuGet.Core/NuGet.Common/PathUtil/PathResolver.cs

```diff
@@ -96,7 +96,7 @@ private static Regex WildcardToRegex(string wildcard)
                     .Replace(@"\?", "."); // ? translates to a single any character
             }
 
-            return new Regex('^' + pattern + '$', RegexOptions.IgnoreCase | RegexOptions.ExplicitCapture);
+            return new Regex('^' + pattern + '$', RegexOptions.IgnoreCase | RegexOptions.ExplicitCapture | RegexOptions.CultureInvariant);
```

Comment:
As noted, you might consider adding this to all your regexes, if you do not use them for culture specific purposes.

Reply:
Applied to all search results provided in the issue.